pkg/trace/obfuscate/sql.go

```diff
@@ -312,18 +313,21 @@ func (oq *ObfuscatedQuery) Cost() int64 {
 
 // attemptObfuscation attempts to obfuscate the SQL query loaded into the tokenizer, using the
 // given set of filters.
-func attemptObfuscation(tokenizer *SQLTokenizer) (*ObfuscatedQuery, error) {
+func attemptObfuscation(tokenizer *SQLTokenizer, options *config.SQLObfuscationConfig) (*ObfuscatedQuery, error) {
```

Comment:
Personal preference and a nit I guess but I wouldn't have been with a pointer for the config.

I would have been with something like:

```go
func attemptObfuscation(tokenizer *SQLTokenizer, options config.SQLObfuscationConfig) (*ObfuscatedQuery, error) {
```

and a `DefaultSQLObfuscationConfig` defining the default values, which would be the object sent by the methods calling with a `nil` in your current code.

Definitely not a strong opinion.

Reply:
Updated this along with your `WithOptions` suggestion, love it, thank you!